Improve deprecation message for ChefShellout cop

Ruchitha-16 opened the pull request:
This change improves the deprecation warning message for the ChefShellout cop.

The previous message mentioned only Mixlib::ShellOut as a replacement. The updated message now clearly recommends either using Mixlib::ShellOut or the shell_out helper, which is a simpler and commonly used alternative.

This update improves clarity for users encountering the deprecation warning without changing any cop behavior or functionality.

Ruchitha-16 committed:
Improve deprecation message for ChefShellout cop

Ruchitha-16 committed:
Sign off ChefShellout deprecation message update  Signed-off-by: Ruchitha <alrljagadam@gmail.com>

# Signed-off update for DCO compliance

Ruchitha-16 commented:
Thanks for reviewing this PR.

Please let me know if any changes are needed from my side.

johnmccrae committed:
Merge branch 'main' into patch-2

dafyddcrosby commented on lib/rubocop/cop/chef/deprecation/chef_shellout.rb:
```diff
@@ -37,7 +37,7 @@ class ChefShellout < Base
           extend AutoCorrector
 
-          MSG = "Don't use the deprecated `Chef::ShellOut` class which was removed in Chef Infra Client 13. Use the `Mixlib::ShellOut` class instead, which behaves identically."
+         MSG = "Avoid using the deprecated `Chef::ShellOut` class (removed in Chef Infra Client 13). Replace it with `Mixlib::ShellOut` or use the `shell_out` helper for simpler command execution."
```
Please remove the spacing change

Ruchitha-16 committed:
Address review feedback for ChefShellout cop

Removed unintended comment and reverted spacing changes as requested by reviewer.

jaymzh reviewed (commented):
This is reasonable, and I appreciate the reference to `shell_out` - but you need DCO.

jaymzh commented:
@Ruchitha-16 - this is a good change, but we need your commits to have sign-off to include it. Can you do that? If not, we'll close next week. Thanks!

jaymzh commented:
No response, gonna close this, but feel free to re-open if you want to finish this.
